Automattic/jetpack
# Protect: Separate scanner and history views via React Router and UI adjustments

Related to https://github.com/Automattic/jetpack/pull/38144

## Proposed changes:

* Adds `/scan/history` and `/scan/history/:filter` routes.
* Adds a `/scan` route and default redirect from `/` -> `/scan`.
* Removes `useThreatHistory()` hook and uses separate routes/root components for the scanner and history views.
* Use initial state provided by server for scan history.
* Only refetch scan history when necessary (i.e. after upgrade or threat fix/ignore).
* Moves filtering of threat history to the client side.
* Adds several supporting components.

### Other information:


## Jetpack product discussion



https://github.com/Automattic/jetpack/pull/38144

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






* Protect Free
  * Ensure no reference/path to threat history is displayed.
* Protect Paid - No Threats
  * Validate "Scan Now" button is displayed on the screen.
  * Review empty History screen.
* Protect Paid - Threats
  * Validate "Scan Now" button is displayed on the screen.
  * Auto-fix and/or ignore threat(s), then navigate to History and validate the new items are displayed.
  * Test reloading the page to validate the server-provided initial state also accurately includes the new threat history.

## Screenshots

<img width="1264" alt="Screenshot 2024-07-12 at 3 48 55 PM" src="https://github.com/user-attachments/assets/a0dc70ce-e412-4e8a-a233-0d6a53632e8f">

<img width="1201" alt="Screenshot 2024-07-12 at 3 49 10 PM" src="https://github.com/user-attachments/assets/68910ebf-d9a0-4d62-9f72-5425afc818cf">

<img width="1223" alt="Screenshot 2024-07-12 at 3 48 23 PM" src="https://github.com/user-attachments/assets/84280c9b-2657-43a9-9da6-84ece5608cf7">
